src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -34,6 +36,31 @@ private class CacheItem
             public TxOut[] OriginalOutputs;
         }
 
+        /// <summary>
+        /// Rewind data with its byte size that will be kept in the queue until it is persisted to a storage
+        /// </summary>
+        private class QueuedRewindData
+        {
+            /// <summary>Rewind data to be saved.</summary>
+            public RewindData RewindData { get; set; }
+
+            /// <summary>The size of the rewind data in bytes.</summary>
```

Comment:
Above set's can be made private.

Reply:
This is just a DTO and not a domain entity and if it needs to be serialised we may end up having issues with private setters